Common/Python/PandasConverter.cs

```diff
@@ -61,149 +61,271 @@ public PandasConverter(PyObject pandas = null)
+    /// </summary>
+    public class PandasData
+    {
+        private Symbol _symbol;
+        private List<DateTime> _timeIndex;
+        private Dictionary<string, List<double>> _series;
+
+        /// <summary>
+        /// Implied levels of a multi index pandas.Series (depends on the security type)
+        /// </summary>
+        public int Levels
+        {
+            get
+            {
+                var levels = 2;
+                if (_symbol.SecurityType == SecurityType.Future) levels += 1;
+                if (_symbol.SecurityType == SecurityType.Option) levels += 3;
+                return levels;
             }
         }
 
         /// <summary>
-        /// Creates the index of pandas.Series
+        /// Initializes an instance of <see cref="PandasData"/> with a sample <see cref="IBaseData"/> object
         /// </summary>
-        /// <param name="symbol"><see cref="Symbol"/> of the security</param>
-        /// <param name="time">Time series axis</param>
-        /// <returns><see cref="PyObject"/> containing a pandas.MultiIndex</returns>
-        private PyObject CreateIndex(Symbol symbol, IEnumerable<DateTime> time)
+        /// <param name="baseData"><see cref="IBaseData"/> object that contains information to be saved in an instance of <see cref="PandasData"/></param>
+        public PandasData(IBaseData baseData)
         {
-            var value = (symbol.HasUnderlying ? symbol.Value : symbol.ToString()).ToPython();
-            var tuples = time.Select(x => new PyTuple(new PyObject[] { value, x.ToPython() }));
-            var names = "symbol,time";
+            var columns = "open,high,low,close";
 
-            if (symbol.SecurityType == SecurityType.Future)
+            if (baseData is TradeBar)
+            {
+                columns += ",volume";
+            }
+            else if (baseData is QuoteBar)
+            {
+                columns += ",askopen,askhigh,asklow,askclose,asksize,bidopen,bidhigh,bidlow,bidclose,bidsize";
+            }
+            else if (baseData is DynamicData)
             {
-                tuples = time.Select(x => new PyTuple(new PyObject[] { symbol.ID.Date.ToPython(), value, x.ToPython() }));
-                names = "expiry," + names;
+                columns = "value," + string.Join(",", ((DynamicData)baseData).GetStorageDictionary().Keys);
             }
 
-            if (symbol.SecurityType == SecurityType.Option)
+            _series = columns.Split(',').ToDictionary(k => k, v => new List<double>());
+            _symbol = baseData.Symbol;
+            _timeIndex = new List<DateTime>();
+        }
+
+        /// <summary>
+        /// Adds an object to the end of the lists
+        /// </summary>
+        /// <param name="baseData"><see cref="IBaseData"/> object that contains information to be saved in an instance of <see cref="PandasData"/></param>
+        public void Add(IBaseData baseData)
+        {
+            _timeIndex.Add(baseData.Time);
```

Comment:
Should we be using time or end time? Philosophical question really. For daily data people would expect time, the date of the data, vs other data end time makes sense. I want the data emitted at 9:42:00 vs saying I want the data that started at 9:39:59

Reply:
Perhaps we need a test of this pandas converter that puts data on and reads it back out and confirms the data is all the same and accessible in the same way